### PR TITLE
Created sec_adcs_certificates.rst

### PR DESCRIPTION
**Pull request by tsluyter**
Jan requested that I describe my process of using ADCS to generate keys and certificates for the Graylog stack. This document is the first iteration of said documentation.

### CODE REVIEW

**Review by tsluyter — COMMENTED** [review]
Looking good! Thanks for the fixes! I cannot approve them in Github, but please go ahead and include these.

File: pages/secure/sec_adcs_certificates.rst [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+If you are going to be generating all the keypairs on your issuing CA or on another management station, then you will need to add the following as well, which will allow you to export the keypair for migration to the Graylog stack servers.
+
+5. Request handling tab:
+  a. Allow the private key to be exported is set to Yes.
+
+
+Generating the keypair and certificates - preparation
+===============
+
+The following instructions assume that you generate all the keypairs on a Windows administrative workstation, or on the issuing CA itself (meaning, you'll need that extra "*Allow the private key to be exported*" flag). You can of course generate all keys on the Graylog stack servers and then simply submit the CSR (certificate signing request) to the CA.
```
Comment (ChristopherKB) [inline pages/secure/sec_adcs_certificates.rst:163]: Change "You can of course generate..." to "You can, of course, generate..."

File: pages/secure/sec_adcs_certificates.rst [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+      _continue=_"dns=graylogreceiveralias&" 
+      _continue=_"ipaddress=192.168.100.105&" 
+
+If you're one of the edge-cases where you will be using an older *Internet Explorer* to talk to the IP address of the host, as opposed to the hostname or its alias, you will need to add::
+
+      _continue=_"dns=192.168.100.105&" 
+
+For some reason IExplore ignores the *ipaddress* field of the SAN (subject alternative name).
+
+The above is only one of the needed .INF files; you will need one for each keypair being generated! So adjust all relevant fields and save each .INF file separately.
```
Comment (ChristopherKB) [inline pages/secure/sec_adcs_certificates.rst:192]: Please bold and italicize the sentence "you will need one for each keypair being generated!"

File: pages/secure/sec_adcs_certificates.rst [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+
+Generating the keypair and certificates - execution
+===============
+
+As said, we're assuming that you're generating the keypairs on your Windows administration station. If you're generating the keypairs on the Graylog Linux hosts, then you will need to use different instructions.
```
Comment (ChristopherKB) [inline pages/secure/sec_adcs_certificates.rst:198]: This sentence is unclear. If you need to use different instructions, where can those instructions be found?

File: pages/secure/sec_adcs_certificates.rst [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+**CA Chain**
+
+Each application requires that you provide the CA chain of your PKI, for inclusion in its trust store. The following assumes that you have one root CA and one issuing CA and that you've put their respective certificates in D:\secrets\graylog::
+
+      openssl x509 -in rootca.crt -outform pem -out D:\secrets\graylog\rootca.pem
+      openssl x509 -in ca.crt -outform pem -out D:\secrets\graylog\ca.pem
+      type D:\secrets\graylog\rootca.pem > D:\secrets\graylog\cachain.pem
+      type D:\secrets\graylog\rootca.pem >> D:\secrets\graylog\cachain.pem
+
+The resulting cachain.pem file can be used in all Graylog stack applications for inclusion in the trust store. You will probably need to run the file through **dos2unix** first though, to fix line endings.
```
Comment (ChristopherKB) [inline pages/secure/sec_adcs_certificates.rst:249]: Might want to expand this. If you will probably need to do it, the guide should cover how.

**Review by ChristopherKB — APPROVED** [review]
Looks good. Still way more confusing than it should be, but the docs are good.

**Comment by tsluyter** [comment]
You've made some great comments Christopher! I'll try and pick all of those up RSN(tm).

File: pages/secure/sec_adcs_certificates.rst (hunk 1 above)
Comment (tsluyter) [reply pages/secure/sec_adcs_certificates.rst:163]:
```suggestion
The following instructions assume that you generate all the keypairs on a Windows administrative workstation, or on the issuing CA itself (meaning, you'll need that extra "*Allow the private key to be exported*" flag). You can, of course, generate all keys on the Graylog stack servers and then simply submit the CSR (certificate signing request) to the CA.
```

File: pages/secure/sec_adcs_certificates.rst (hunk 2 above)
Comment (tsluyter) [reply pages/secure/sec_adcs_certificates.rst:192]:
```suggestion
The above is only one of the needed .INF files; **you will need one for each key pair being generated**! So adjust all relevant fields and save each .INF file separately.
```

File: pages/secure/sec_adcs_certificates.rst (hunk 3 above)
Comment (tsluyter) [reply pages/secure/sec_adcs_certificates.rst:198]:
```suggestion
As said, we're assuming that you're generating the keypairs on your Windows administration station. If you're generating the key pairs on the Graylog Linux hosts, then you will need to use OpenSSL instructions (`For example <https://www.digitalocean.com/community/tutorials/openssl-essentials-working-with-ssl-certificates-private-keys-and-csrs>`_). 
```

File: pages/secure/sec_adcs_certificates.rst (hunk 4 above)
Comment (tsluyter) [reply pages/secure/sec_adcs_certificates.rst:249]:
```suggestion
The resulting cachain.pem file can be used in all Graylog stack applications for inclusion in the trust store. After uploading the file to Linux, you will need to run the file through **dos2unix** first, to fix line endings.
```

**Comment by tsluyter** [comment]
Hey @ChristopherKB I've added suggestions for each of your comments. Thanks for your time and help.

**Comment by tsluyter** [comment]
Wait, I just realized something important. Ben on the forums pointed out that the PKCS#8 keys are not usable for inputs on Graylog 3.0. Apparently they work differently. So these instructions are 100% guaranteed for <= v2.5. Inputs of 3.0 apparently need additional work. 

EDIT:
Maybe I misunderstood Ben. Will need to investigate. 

[See here.](https://community.graylog.org/t/a-few-observations-on-upgrading-from-2-5-to-3-0)

**Pull request merged**